src/main/java/com/linkedin/xinfra/monitor/services/MultiClusterTopicManagementService.java

```diff
@@ -426,13 +431,12 @@ void maybeReassignPartitionAndElectLeader() throws Exception {
               "Configured replication factor {} is smaller than the current replication factor {} of the topic {} in cluster.",
               _replicationFactor, currentReplicationFactor, _topic);
 
-        if (expectedReplicationFactor > currentReplicationFactor && !zkClient
-            .reassignPartitionsInProgress()) {
+        if (expectedReplicationFactor > currentReplicationFactor
+            && Utils.ongoingPartitionReassignments(_adminClient).isEmpty()) {
```

Comment:
Can we check the ongoing partition reassignments only for this single topic?

Reply:
I mean passing the list of partitions, and call the API 
listPartitionReassignments(Set<TopicPartition> partitions)

Reply:
We can. We wanted to replicate the previous behavior of `.reassignPartitionsInProgress()` which checks for all. 
Do we want to check only a set of topic partitions? @gitlw

Reply:
I think you raise a very good point but in order to keep the exact same behaviour to be on a much safer side, we should probably keep the Utils.ongoingPartitionReassignments.isEmpty? — which is similar to  !zkClient.reassignPartitionsInProgress()
Let's check if getting all of the ongoing partition reassignments is costly when using the new API.